hack/machineset.sh

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env bash
+
+# machineset.sh - apply/delete the Windows MachineSet for Windows Machine Config Operator
+#
+# USAGE
+#    machineset.sh apply/delete
+# OPTIONS
+#    $1      Action       apply/delete the MachineSet
+# PREREQUISITES
+#    oc                   to apply MachineSets on the cluster(cluster should be logged in)
+#    aws                  to fetch Windows AMI id for AWS platform (only required for clusters running on AWS)
+
+ACTION=$1
+if [[ ! "$ACTION" =~ ^apply|delete$ ]]; then
+    error-exit "Action (1st parameter) must be \"apply\" or \"delete\""
+fi
+
+# Creates a MachineSet for AWS Cloud Provider
+get_aws_ms(){
```

Comment:
Having this per provider is going to hard to maintain given the majority of the spec is the same across providers. How about a single function that takes the provider as input and generates the yaml differently based on the provider?

Reply:
The only thing we are actually changing for the cloud-provider is the `provider spec` we can keep the rest of `yaml` as it is for all cloud providers... 
But except `user-data` part in provider spec, rest everything is different for different cloud providers. I mean, the code duplication can be reduced in this case, but i think we might still have to have these functions for different cloud providers to make the code look good and understandable, but i am open for suggestions ,,,

Reply:
Sure the `providerSpec` will be different, however the rest should be a common template where you replace things like `infraID` etc. I am not averse to having different functions for each provider but there needs to be more in common.

This can also be tackled in a follow up PR.

Reply:
I'll reduce the code duplication in this PR. Wont take much effort.. 
Rest of the things can be done in follow up PRs